packages/access-client/src/agent-data.js

```diff
@@ -143,4 +145,18 @@ export class AgentData {
     this.delegations.delete(cid.toString())
     await this.#save(this.export())
   }
+
+  /**
+   * The current session proof.
+   */
+  sessionProof() {
```

Comment:
This is written in a way that seems to imply there will be a zero or one 'session proof' that can be determine at all times, which I think may be an invalid assumption. The session proof you need is a function of the action your atrying to take.

Reply:
Put it differently you'll have a set of attestations (sessions) that which have `capabilities[0].nb.proof` CID corresponding to specific delegation. Idea is to send delegation and a session together, so perhaps better api would be to a function that takes `delegation` and then finds `session` where `session.capabilities[0].proof.toString() === delegation.cid.toString()`

Reply:
Maybe we could make this `sessionProof` `protected` for now and that way we can merge this and iterate on the api. @travis do you need this to be `public`? i.e. do you call it directly from w3up-client et al?

Reply:
don't need to call it from w3up-client nope, let's make it `protected`! I think this was just something @alanshaw threw in provisionally so definitely feel free to change it

Reply:
removed `sessionProof` in https://github.com/web3-storage/w3protocol/pull/545/files